sklearn/feature_selection/from_model.py

```diff
@@ -68,11 +71,16 @@ def transform(self, X, threshold=None):
 
         # Retrieve threshold
         if threshold is None:
-            if hasattr(self, "penalty") and self.penalty == "l1":
+            threshold = getattr(self, "threshold", None)
+
+        if threshold is None:
+            # Lasso has a l1 penalty but no penalty param.
+            if (hasattr(self, "penalty") and self.penalty == "l1" or
+                'Lasso' in self.__class__.__name__):
```

Comment:
> I meant that even Non-transformers, can have transform methods.

IMO, that's a contradiction, effectively. By implementing fit() and
transform(), they implement the Transformer API, and are thus transformers.
The common tests should be explicitly non-mutually-exclusive between
predictors and transformers. What sorts of tests were failing?

Having said that, I'm not sure I agree with this design and might rather
see this style of feature selection come through a wrapping meta-estimator
than a mixin.

On 12 February 2015 at 17:40, Manoj Kumar <notifications@github.com> wrote:

>  In sklearn/feature_selection/from_model.py
> <https://github.com/scikit-learn/scikit-learn/pull/4241#discussion_r24561488>
> :
>
> > @@ -68,11 +71,16 @@ def transform(self, X, threshold=None):
> >
> >          # Retrieve threshold
> >          if threshold is None:
> > -            if hasattr(self, "penalty") and self.penalty == "l1":
> > +            threshold = getattr(self, "threshold", None)
> > +
> > +        if threshold is None:
> > +            # Lasso has a l1 penalty but no penalty param.
> > +            if (hasattr(self, "penalty") and self.penalty == "l1" or
> > +                'Lasso' in self.__class__.__name__):
>
> Should it default to median for ENet?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/4241/files#r24561488>.
>